src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -53,35 +63,439 @@ static ColliderCylinderInit D_809B22D0 = {
+        }
+    }
+}
+
+void EnAnubice_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 zero;
+    BgHidanCurtain* flameCircle;
+    s32 i;
+    Vec3f sp48;
+    Vec3f sp3C;
+    EnAnubice* this = THIS;
+
+    if (this->actionFunc != EnAnubice_SetupDie && this->actionFunc != EnAnubice_Die &&
+        this->actor.shape.yOffset == 0.0f) {
+        EnAnubice_Hover(this, globalCtx);
+        for (i = 0; i < 5; i++) {
+            flameCircle = this->flameCircles[i];
+
+            if (flameCircle != NULL && fabsf(flameCircle->actor.world.pos.x - this->actor.world.pos.x) < 60.0f &&
+                fabsf(this->flameCircles[i]->actor.world.pos.z - this->actor.world.pos.z) < 60.0f &&
+                flameCircle->timer != 0) {
+                Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_PROP);
+                this->actor.flags &= ~1;
+                func_80032C7C(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_DEAD);
+                this->actionFunc = EnAnubice_SetupDie;
+                return;
+            }
+        }
+
+        if (this->col.base.acFlags & 2) {
+            this->col.base.acFlags &= ~2;
+            if (this->actor.colChkInfo.damageEffect == 2) {
+                Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_PROP);
+                this->actor.flags &= ~1;
+                func_80032C7C(globalCtx, &this->actor);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANUBIS_DEAD);
+                this->actionFunc = EnAnubice_SetupDie;
+                return;
+            }
+
+            if (!this->isKnockedback) {
+                this->knockbackTimer = 10;
+                this->isKnockedback = true;
+
+                sp48.x = 0.0f;
+                sp48.y = 0.0f;
+                sp48.z = -10.0f;
+                sp3C.x = 0.0f;
+                sp3C.y = 0.0f;
+                sp3C.z = 0.0f;
+
+                Matrix_RotateY(BINANG_TO_RAD(this->actor.shape.rot.y), MTXMODE_NEW);
+                Matrix_MultVec3f(&sp48, &sp3C);
+
+                this->actor.velocity.x = sp3C.x;
+                this->actor.velocity.z = sp3C.z;
+                this->knockbackRecoveryVelocity.x = -sp3C.x;
+                this->knockbackRecoveryVelocity.z = -sp3C.z;
+
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_NUTS_CUTBODY);
+            }
+        }
+
+        if (this->isKnockedback) {
+            this->actor.shape.rot.y += 6500;
+            Math_ApproachF(&this->actor.velocity.x, this->knockbackRecoveryVelocity.x, 0.3f, 1.0f);
+            Math_ApproachF(&this->actor.velocity.z, this->knockbackRecoveryVelocity.z, 0.3f, 1.0f);
+
+            zero = 0.0f;
+            if (zero) {}
+
+            if (this->knockbackTimer == 0) {
+                this->actor.velocity.x = this->actor.velocity.z = 0.0f;
+                this->knockbackRecoveryVelocity.x = this->knockbackRecoveryVelocity.z = 0.0f;
```

Comment:
does chaining all of these into one help with the strange zero issue above? 
if not you dont need to chain them all together, the line would be a bit long

Reply:
I messed with it and it doesn't help. Chaining here makes sense because the o.g. code likely did it.

Reply:
the way he has it is best imo. chaining all 4 of those is a bit obnoxious, so if it doesnt help with matching id say to keep it how it is